app/src/main/java/com/odysee/app/ui/findcontent/FileViewFragment.java

```diff
@@ -2394,44 +2437,76 @@ public void onError(Exception error) {
         }
     }
 
-    private void handleMainActionForClaim() {
-        if (claim.isFree() && ((claim.isPlayable() && Lbryio.isSignedIn()) || !claim.hasSource())) {
-            // always use lbry.tv streaming when signed in and playabble
-            startTimeMillis = System.currentTimeMillis();
-            showExoplayerView();
-            playMedia();
+    private void resolvePlaylistClaimsAndPlayFirst() {
+        if (playlistResolved) {
             return;
         }
 
-        if (Lbry.SDK_READY) {
-            // Check if the file already exists for the claim
-            if (claim.getFile() != null) {
-                playOrViewMedia();
-            } else {
-                // check if the file exists from file list
-                boolean saveFile = downloadRequested || !claim.isPlayable();
-                if (!saveFile) {
-                    startTimeMillis = System.currentTimeMillis();
-                    fileGet(false);
-                    return;
-                } else {
-                    tryOpenFileOrFileGet();
+        Helper.setViewVisibility(layoutLoadingState, View.VISIBLE);
+        Helper.setViewVisibility(layoutResolving, View.VISIBLE);
+        Helper.setViewVisibility(layoutNothingAtLocation, View.GONE);
+        Helper.setViewVisibility(layoutDisplayArea, View.GONE);
+
+        Map<String, Object> options = new HashMap<>();
+        options.put("claim_type", "stream");
+        options.put("page", 1);
+        options.put("page_size", 999);
+        options.put("claim_ids", claim.getClaimIds());
+        ClaimSearchTask task = new ClaimSearchTask(options, Lbry.API_CONNECTION_STRING, null, new ClaimSearchResultHandler() {
+            @Override
+            public void onSuccess(List<Claim> claims, boolean hasReachedEnd) {
+                playlistResolved = true;
+
+                // reorder the claims based on the order in the list, TODO: find a more efficient way to do this
+                Map<String, Claim> playlistClaimMap = new LinkedHashMap<>();
+                for (String id : claim.getClaimIds()) {
```

Comment:
claim.getClaimIds() is not expected to change through the loop execution, but this code will still make the call every time. I suggest you to store it in a variable outside the loop and use it, instead. This way, the app will be accessing the memory position of the object instead of recreating the object every pass on the loop.